### PR TITLE
CO-1804 v2 design

### PR DESCRIPTION
**Pull request by tsoliangwu0130**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -12,106 +12,206 @@ info:
...
+            type: string
+            format: date
+            example: '2018-12-31'
+        - name: filter[name]
+          in: query
+          description: Search by location name
+          required: false
+          schema:
+            type: string
+            example: Corvallis
```
Comment (Ruefa) [inline openapi.yaml:136]: Some of these need examples. I think this one should go on `city` and a different one put on `name`

File: openapi.yaml [hunk 2]
```diff
@@ -126,25 +226,15 @@ components:
...
+      description: Search by 4-character business center name
+      required: true
       schema:
-        type: integer
-        minimum: 1
-        maximum: 500
-        default: 25
+        type: string
+        example: ABCD
+        pattern: ^.{4}$
```
Comment (Ruefa) [inline openapi.yaml]: I feel this should be stricter than just `.`. You can do `[a-zA-Z0-9]` for any alphanumerical character.

File: openapi.yaml [hunk 3]
```diff
@@ -158,162 +248,227 @@ components:
...
-          format: uri
-          nullable: true
-          description: The previous page of data
-        next:
-          type: string
-          format: uri
-          nullable: true
-          description: The next page of data
-    PetResource:
+    PositionResource:
```
Comment (Ruefa) [inline openapi.yaml:266]: `type: object`

File: openapi.yaml [hunk 4]
```diff
@@ -158,162 +248,227 @@ components:
...
-                  format: float
-                  description: Length in inches
-                  example: 18
-                chest:
+                code:
+                  type: string
+                  nullable: false
+                  description: Organization code for the department
+                  example: '230600'
+                title:
```
Comment (Ruefa) [inline openapi.yaml]: Should we be consistent with persons and use `description` instead of `title`?

File: openapi.yaml [hunk 5]
```diff
@@ -158,162 +248,227 @@ components:
...
+                  type: string
+                  description: National occupation code
+                  example: '1234'
+                  nullable: false
+                description:
+                  type: string
+                  description: Description of national occupation code
+                  example: Colleges/Schools Professionals
+                  nullable: false
+    PositionSetResult:
```
Comment (Ruefa) [inline openapi.yaml:336]: `type: object`

File: openapi.yaml [hunk 6]
```diff
@@ -158,162 +248,227 @@ components:
...
+                  type: string
+                  nullable: false
+                  description: Organization title for the department
+                  example: FOR - Forestry Administration
+            businessCenter:
+              type: string
+              description: Name of the business center
+              example: FOBC
+              nullable: false
+    DepartmentSetResult:
```
Comment (Ruefa) [inline openapi.yaml:384]: `type: object`

**Review by Ruefa — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 1 above)
Comment (tsoliangwu0130) [reply openapi.yaml:136]: Well, I think the name for Corvallis location is also Corvallis and I think it's very common that name is the same with city or state. This is what I got from v1:

```json
{
    "id": "02A",
    "type": "Location",
    "attributes": {
        "name": "Corvallis",
        "city": "Corvallis",
        "county": "Benton County",
        "state": "Oregon",
        "stateCode": "OR",
        "minimumWageClassification": "Standard",
        "minimumWage": 11.25
    },
    "links": {
        "self": "https://oregonstateuniversity-test.apigee.net/v1/hr/locations/02A?date=2020-05-19"
    }
},
```

File: openapi.yaml (hunk 2 above)
Comment (tsoliangwu0130) [reply openapi.yaml]: Good idea.

File: openapi.yaml (hunk 4 above)
Comment (tsoliangwu0130) [reply openapi.yaml]: Agree.

**Review by Ruefa — APPROVED** [review]

**Pull request merged**